packages/flutter/lib/src/rendering/view.dart

```diff
@@ -38,6 +38,14 @@ class ViewConfiguration {
     return Matrix4.diagonal3Values(devicePixelRatio, devicePixelRatio, 1.0);
   }
 
+  /// Whether the root render object needs to replace the root transform layer.
+  ///
+  /// By default, the root layer will only be replaced when [devicePixelRatio]
```

Comment:
Document that this must return true if this.toMatrix and oldConfiguration.toMatrix return different values, and false otherwise.

Reply:
I'm wondering if there will be a possibility in the future that the matrix has changed, but the root layer does not need to be replaced. Based on this, I disassembled `_updateMatricesAndCreateNewRootLayer`. What is your thoughts about this?

Reply:
Additionally, if we must replace the root layer when the matrix changes, maybe we should not expose this API to folks.

Reply:
Currently, the only input to the root layer is the matrix retrieved from `toMatrix`, so let's just design this for the current use case. If this changes in the future, we will have to re-think this anyways.

> maybe we should not expose this API to folks.

As an alternative to exposing this new API we could just compare the current _rootTransform with the matrix obtained from the new configuration and only replace the layer if they are different. That sounds reasonable to me as well.

Reply:
Done